src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -485,6 +485,28 @@ def is_signature_valid(self):
 
         return True
 
+    @property
+    def challenge_password(self):
+        obj = _txt2obj_gc(
+            self._backend, x509.AttributeOID.CHALLENGE_PASSWORD.dotted_string
+        )
+        pos = self._backend._lib.X509_REQ_get_attr_by_OBJ(
+            self._x509_req, obj, -1
+        )
+        if pos == -1:
+            return None
+        else:
+            attr = self._backend._lib.X509_REQ_get_attr(self._x509_req, pos)
+            self._backend.openssl_assert(attr != self._backend._ffi.NULL)
+            asn1_type = self._backend._lib.X509_ATTRIBUTE_get0_type(attr, pos)
+            self._backend.openssl_assert(asn1_type != self._backend._ffi.NULL)
+            data = self._backend._lib.X509_ATTRIBUTE_get0_data(
+                attr, pos, asn1_type.type, self._backend._ffi.NULL
+            )
+            self._backend.openssl_assert(data != self._backend._ffi.NULL)
+            data = self._backend._ffi.cast("ASN1_STRING *", data)
```

Comment:
Is this cast safe? What happens if I put a challenge password OID + something that's not a string, e.g. an integer or a sequence or anything else.

Reply:
We can avoid a problem with that by asserting on the return of `X509_ATTRIBUTE_get0_type`

Reply:
I hate to say it... but please write a test for this, it'd be extremely dangerous if this bug occurred.

(der2ascii + ascii2der are your friends)

Reply:
![image](https://user-images.githubusercontent.com/161495/61599979-0249a580-abfc-11e9-83f4-39a2b8e75db7.png)